File: frontend/projects/api/src/lib/models/workout-dto.ts
import { EquipmentType } from './equipment-type';

/** A logged workout. `startedAt` is an ISO UTC timestamp. */
export interface WorkoutDto {
  readonly id: string;
  readonly equipment: EquipmentType;
  readonly startedAt: string;
  readonly durationMinutes: number;
  readonly distanceKm: number | null;
  readonly avgHeartRateBpm: number | null;
  readonly activeCalories: number | null;
  readonly notes: string | null;
}

File: backend/src/Shalom.Application/Contracts/WorkoutDto.cs
using Shalom.Domain.Enums;

namespace Shalom.Application.Contracts;

/// <summary>A logged workout. <paramref name="Equipment"/> serializes as its enum name (e.g. "IndoorBike").</summary>
public record WorkoutDto(
    Guid Id,
    EquipmentType Equipment,
    DateTimeOffset StartedAt,
    int DurationMinutes,
    decimal? DistanceKm,
    int? AvgHeartRateBpm,
    int? ActiveCalories,
    string? Notes);
